pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -174,7 +175,8 @@ public String consumerName() {
                         .build();
 
                 ByteBuf metadataAndPayload = entry.getDataBuffer();
-
+                // increment ref-count of data and release at the end of process: so, we can get chance to call entry.release
+                entry.retain();
```

Comment:
yes, that's correct. So, if we remove `release()` then we have to call `entry.data.release()` and `entry.recycle()` explicitly. So, we have two options:

1. Every time make two explicit call
```
entry.getDataBuffer().release();
entry.recycle();
```

2. rename `release()` with `releaseAndRecycle()` which clarifies purpose => release data and recycle entry. and caller doesn't have to call multiple things.

Do, you think we 2nd option will solve this concern or we should go with the 1st one.